conanfile.py

```diff
@@ -1,112 +1,81 @@
+
+        cmake_file = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmake_file, "project(MONGO_CXX_DRIVER LANGUAGES CXX)", conan_magic_lines)
         cmake = CMake(self)
-        if self.settings.compiler == 'Visual Studio':
-            cmake.definitions["BSONCXX_POLY_USE_BOOST"] = 1 # required for Windows.
-        cmake.configure(source_dir="sources")
+        cmake.definitions["BSONCXX_POLY_USE_MNMLSTC"] = self.options.polyfill == "mnmlstc"
+        cmake.definitions["BSONCXX_POLY_USE_STD_EXPERIMENTAL"] = self.options.polyfill == "experimental"
+        cmake.definitions["BSONCXX_POLY_USE_BOOST"] = self.options.polyfill == "boost"
+        cmake.configure(source_dir=self._source_subfolder)
         cmake.build()
 
-    def purge(self, dir, pattern):
-        for f in os.listdir(dir):
-            if re.search(pattern, f):
-                # print "removing {0}".format(os.path.join(dir, f))
-                os.remove(os.path.join(dir, f))
-
     def package(self):
-        self.copy(pattern="COPYING*", src="sources")
-        self.copy(pattern="*.hpp", dst="include/bsoncxx", src="sources/src/bsoncxx", keep_path=True)
-        self.copy(pattern="*.hpp", dst="include/mongocxx", src="sources/src/mongocxx", keep_path=True)
-        self.copy(pattern="*.hpp", dst="include/bsoncxx", src="src/bsoncxx", keep_path=True)
-        self.copy(pattern="*.hpp", dst="include/mongocxx", src="src/mongocxx", keep_path=True)
-        self.copy(pattern="*.hpp", dst="include/bsoncxx/third_party/mnmlstc/core", src="src/bsoncxx/third_party/EP_mnmlstc_core-prefix/src/EP_mnmlstc_core/include/core", keep_path=False)
-        # self.copy(pattern="*.dll", dst="bin", src="bin", keep_path=False)
-
-        # self.purge("lib", "lib.*testing.*".format(self.version))
-        # self.purge("lib", "lib.*mocked.*".format(self.version))
-        # # self.purge("lib", "lib.*_noabi.*".format(self.version))
-
-        try:
-            os.rename("lib/libmongocxx-static.a", "lib/libmongocxx.a")
-        except:
-            pass
-        try:
-            os.rename("lib/libbsoncxx-static.a", "lib/libbsoncxx.a")
-        except:
-            pass
-        try:
-            os.rename("lib/libmongocxx-static.lib", "lib/libmongocxx.lib")
-        except:
-            pass
-        try:
-            os.rename("lib/libbsoncxx-static.lib", "lib/libbsoncxx.lib")
-        except:
-            pass
-        self.copy(pattern="lib*cxx.lib", dst="lib", src="lib", keep_path=False)
-        self.copy(pattern="lib*cxx.a", dst="lib", src="lib", keep_path=False)
-        self.copy(pattern="lib*cxx.so*", dst="lib", src="lib", keep_path=False)
-        self.copy(pattern="lib*cxx.dylib", dst="lib", src="lib", keep_path=False)
-        self.copy(pattern="lib*cxx._noabi.dylib", dst="lib", src="lib", keep_path=False)
+        # Do not reconfigure because that causes a full rebuild
+        CMake(self).install()
```

Comment:
Use [_configure_cmake](https://github.com/bincrafters/conan-basis_universal_installer/blob/325426dd68dbe7f372679918a0c080731ef1b53c/conanfile.py#L28) instead of calling CMake directly twice.

Reply:
I would be using the pattern without following the pattern since I can't call `configure` in `_configure_cmake` without causing a full rebuild. Are you sure that's better? At least this way I am very explicit about not following the pattern

Reply:
> I don't know why a full rebuild is caused by reconfiguring

because you calling CMake twice

You can avoid it by declaring a common variable for cmake. Take a look on this example: https://github.com/conan-io/conan-center-index/blob/59f0e51e02010e0e593e2cd22be62ee756dedac9/recipes/libwebp/all/conanfile.py#L22

Reply:
Well, not necessarily. Reconfiguring an already configured and built CMake project does not always imply it gets rebuilt. I'll follow the member variable variant you linked, though.